openwisp_utils/qa.py

```diff
@@ -181,6 +185,38 @@ def check_commit_message():
         sys.exit(1)
 
 
+def check_ReST():
+    settings_overrides = {}
+    settings_overrides.update(readme_rst.SETTINGS)
+    settings_overrides['report_level'] = 2
+    files = [_ for _ in iglob('./**/*.rst', recursive=True)]
+    errors = []
+    for file in files:
+        data = read_ReST_file(file)
+        try:
+            docutils.core.publish_string(data, settings_overrides=settings_overrides)
+            clean_body = readme_rst.clean(''.join(data))
+            if clean_body is None:
+                raise ValueError("Output Failed")
+        except Exception as e:
+            errors.append(e)
+    if len(errors):
+        body = (
+            'There are some Errors in your RST file syntax'
+            'which can create problems in future:\n\n'
+        )
+        for error in errors:
+            body += '- {}\n'.format(error)
+        print(body)
+        sys.exit(1)
```

Comment:
please write a few automated tests for this code to ensure future contributors can change it without risking to break the main functionality

Reply:
Pardon, I am not able to understand which type of test cases.

Reply:
Tests are here: https://github.com/openwisp/openwisp-utils/blob/master/tests/test_project/tests/test_qa.py